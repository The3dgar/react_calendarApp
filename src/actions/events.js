import Swal from "sweetalert2";
import { fetchWithToken } from "../helper/fetch";
import { prepareEvents } from "../helper/prepareEvents";
import { types } from "../types/types";

// disparar si se graba en la base de datos
export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;
    try {
      const resp = await fetchWithToken("/events", event, "POST");
      const body = await resp.json();

      if (body.ok) {
        event.id = body.event.id;
        event.user = {
          _id: uid,
          name,
        };
        dispatch(eventAddNew(event));
      }
    } catch (error) {
      console.log(error);
    }
  };
};
const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event,
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event,
});

export const eventClearActiveEvent = () => ({
  type: types.eventClearActiveEvent,
});

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken("events");
      const body = await resp.json();
      console.log(body);
      const events = prepareEvents(body.events);
      dispatch(eventLoaded(events));
    } catch (error) {
      console.log(error);
    }
  };
};

const eventLoaded = (events) => ({
  type: types.eventLoaded,
  payload: events,
});

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken(`/events/${event.id}`, event, "PUT");
      const body = await resp.json();
      if (body.ok) {
        dispatch(eventUpdated(event));
      } else {
        Swal.fire("Error", body.msg, "error");
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event,
});

export const eventStartDelete = (event) => {
  return async (dispatch, getState) => {
    const { activeEvent: event } = getState().calendar;
    try {
      const resp = await fetchWithToken(`/events/${event.id}`, event, "DELETE");
      const body = await resp.json();
      if (body.ok) {
        dispatch(eventDeleted(event));
      } else {
        Swal.fire("Error", body.msg, "error");
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventDeleted = () => ({
  type: types.eventDeleted,
});

export const eventLogout = () => ({
  type: types.eventLogout
})
