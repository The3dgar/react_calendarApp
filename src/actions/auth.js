import Swal from "sweetalert2";
import { fetchWithOutToken, fetchWithToken } from "../helper/fetch";
import { types } from "../types/types";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  // dispatch gracias al thunk
  return async (dispatch) => {
    const resp = await fetchWithOutToken("auth", { email, password }, "POST");
    const body = await resp.json();

    if (body.ok) {
      const { token, uid, name } = body;
      localStorage.setItem("token", token);
      localStorage.setItem("token_init", new Date().getTime());

      dispatch(login({ uid, name }));
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  };
};

const login = (user) => ({
  type: types.authLogin,
  payload: user,
});

export const startRegister = (email, password, name) => {
  return async (dispatch) => {
    const resp = await fetchWithOutToken(
      "auth/new",
      { email, password, name },
      "POST"
    );
    const body = await resp.json();

    if (body.ok) {
      const { token, uid, name } = body;
      localStorage.setItem("token", token);
      localStorage.setItem("token_init", new Date().getTime());

      dispatch(login({ uid, name }));
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startLogout = () => {
  return dispatch => {
    localStorage.clear()
    dispatch(eventLogout())
    dispatch(logout())
  }
}

const logout = () => ({type: types.authLogout})

export const startChecking  = ()  => {
  return async dispatch => {
    const token = localStorage.getItem("token")

    if(!token) return dispatch(checkingFinish())
    
    const resp = await fetchWithToken("auth/renew");
    const body = await resp.json();
    if (body.ok) {
      const { token, uid, name } = body;
      localStorage.setItem("token", token);
      localStorage.setItem("token_init", new Date().getTime());

      dispatch(login({ uid, name }));
    } else {
      Swal.fire("Errores", body.msg, "error");
      dispatch(checkingFinish())
    }
  }
}

const checkingFinish = () => ({
  type: types.authCheckingFinish
})