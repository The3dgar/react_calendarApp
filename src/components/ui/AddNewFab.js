import React from 'react'
import { useDispatch } from 'react-redux'
import { uiOpenModal } from '../../actions/ui'

export const AddNewFab = () => {
  const dispatch = useDispatch()
  const handleOpenModal = () => {
    dispatch(uiOpenModal())
  }
  return (
    <button className="btn btn-primary fab" onClick={handleOpenModal}>
      <i className="fas fa-plus"></i>
    </button>
  )
}
