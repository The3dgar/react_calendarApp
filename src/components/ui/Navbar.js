import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../../actions/auth";

export const Navbar = () => {
  const { name } = useSelector((state) => state.auth);

  const dispatch = useDispatch()
  const handleLogout = () => {

    dispatch(startLogout())
  };
  return (
    <div className="navbar navbar-dark bg-dark mb-4">
      <span className="navbar-brand">{name}</span>

      <button className="btn btn-outline-danger" onClick={handleLogout}>
        <i className="fas fa-sign-out-alt"></i> Salir
      </button>
    </div>
  );
};
