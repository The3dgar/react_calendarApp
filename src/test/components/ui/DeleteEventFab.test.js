import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from "redux-mock-store"
import thunk from "redux-thunk"
import "@testing-library/jest-dom"
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

jest.mock( "../../../actions/events", () => ({
  eventStartDelete: jest.fn()
}))

const middlewares  = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}
const store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab/>
  </Provider>
) 

describe('testing in <DeleteEventFab/>', () => {
  
  test('should snap correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should dispatch eventStartDelete on click', () => {
    wrapper.find("button").prop("onClick")()

    expect(eventStartDelete).toHaveBeenCalled()
  })
  
  
})
