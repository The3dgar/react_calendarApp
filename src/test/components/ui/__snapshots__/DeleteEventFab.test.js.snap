// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in <DeleteEventFab/> should snap correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
</button>
`;
